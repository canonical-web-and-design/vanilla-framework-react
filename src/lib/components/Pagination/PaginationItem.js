import React from 'react';
import PropTypes from 'prop-types';

const PaginationItem = (props) => {
  const {
    href, label, next, previous, title,
  } = props;

  let className = 'p-pagination__link';

  if (next) {
    className += '--next';
  } else if (previous) {
    className += '--previous';
  }

  return (
    <a className={className} href={href}>
      <span className="p-pagination__label">{label}</span>
      <span className="p-pagination__title">{title}</span>
    </a>
  );
};

PaginationItem.defaultProps = {
  href: null,
  label: '',
  next: false,
  previous: false,
  title: '',
};

PaginationItem.propTypes = {
  href: PropTypes.string,
  label: PropTypes.string,
  next: PropTypes.bool,
  previous: PropTypes.bool,
  title: PropTypes.string,
};

PaginationItem.displayName = 'PaginationItem';

export default PaginationItem;
