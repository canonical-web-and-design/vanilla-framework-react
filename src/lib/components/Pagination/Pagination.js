import React from 'react';

import PaginationItem from './PaginationItem';

const Pagination = props => (
  <div className="p-pagination">
    { props.children }
  </div>
);

Pagination.defaultProps = {
  children: null,
};

Pagination.propTypes = {
  children: (props, propName, componentName) => {
    const prop = props[propName];
    let error = null;

    React.Children.forEach(prop, (child) => {
      if (child.type !== PaginationItem) {
        error = new Error(`${componentName} children should be of type "PaginationItem".`);
      }
    });

    return error;
  },
};

Pagination.displayName = 'Pagination';

export default Pagination;
