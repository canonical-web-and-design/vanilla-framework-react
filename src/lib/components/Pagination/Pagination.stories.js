import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, text } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';

import Pagination from './Pagination';
import PaginationItem from './PaginationItem';

storiesOf('Pagination', module)
  .add('Next',
    withInfo('The Pagination component should be used to navigate from one article to the next or previous in a chronological fashion.')(() => (
      <Pagination>
        <PaginationItem
          next={boolean('Next', true)}
          previous={boolean('Previous', false)}
          href={text('href', '')}
          label={text('Label', 'Next')}
          title={text('Title', 'Consectetur adipisicing elit')}
        />
      </Pagination>),
    ),
  )

  .add('Previous',
    withInfo('info')(() => (
      <Pagination>
        <PaginationItem
          next={boolean('Next', false)}
          previous={boolean('Previous', true)}
          href={text('href', '')}
          label={text('Label', 'Previous')}
          title={text('Title', 'Lorem ipsum dolor sit amet')}
        />
      </Pagination>),
    ),
  )

  .add('Both',
    withInfo('info')(() => (
      <Pagination>
        <PaginationItem
          previous
          href={text('Prev href', '')}
          label={text('Prev Label', 'Previous')}
          title={text('Prev Title', 'Lorem ipsum dolor sit amet')}
        />
        <PaginationItem
          next
          href={text('Next href', '')}
          label={text('Next Label', 'Next')}
          title={text('Next Title', 'Consectetur adipisicing elit')}
        />
      </Pagination>),
    ),
  );
