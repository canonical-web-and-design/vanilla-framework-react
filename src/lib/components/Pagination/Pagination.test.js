import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import Pagination from './Pagination';
import PaginationItem from './PaginationItem';

describe('<Pagination>', () => {
  it('renders default Pagination with PaginationItem correctly', () => {
    const pagination = ReactTestRenderer.create(
      <Pagination>
        <PaginationItem />
      </Pagination>,
    );
    const json = pagination.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders PaginationItem with next prop correctly', () => {
    const pagination = ReactTestRenderer.create(
      <Pagination>
        <PaginationItem next />
      </Pagination>,
    );
    const json = pagination.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders PaginationItem with previous prop correctly', () => {
    const pagination = ReactTestRenderer.create(
      <Pagination>
        <PaginationItem previous />
      </Pagination>,
    );
    const json = pagination.toJSON();
    expect(json).toMatchSnapshot();
  });
});
