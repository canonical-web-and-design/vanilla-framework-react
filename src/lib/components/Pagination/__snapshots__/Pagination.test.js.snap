// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination> renders PaginationItem with next prop correctly 1`] = `
<div
  className="p-pagination"
>
  <a
    className="p-pagination__link--next"
    href={null}
  >
    <span
      className="p-pagination__label"
    >
      
    </span>
    <span
      className="p-pagination__title"
    >
      
    </span>
  </a>
</div>
`;

exports[`<Pagination> renders PaginationItem with previous prop correctly 1`] = `
<div
  className="p-pagination"
>
  <a
    className="p-pagination__link--previous"
    href={null}
  >
    <span
      className="p-pagination__label"
    >
      
    </span>
    <span
      className="p-pagination__title"
    >
      
    </span>
  </a>
</div>
`;

exports[`<Pagination> renders default Pagination with PaginationItem correctly 1`] = `
<div
  className="p-pagination"
>
  <a
    className="p-pagination__link"
    href={null}
  >
    <span
      className="p-pagination__label"
    >
      
    </span>
    <span
      className="p-pagination__title"
    >
      
    </span>
  </a>
</div>
`;
